stm/stm.ml

```diff
@@ -1979,13 +1979,15 @@ end = struct (* {{{ *)
   let vernac_interp ~solve ~abstract ~cancel_switch nworkers priority safe_id id
     { indentation; verbose; expr = e; strlen } : unit
   =
-    let e, time, batch, fail =
-      let rec find ~time ~batch ~fail v = CAst.with_loc_val (fun ?loc -> function
-        | VernacTime (batch,e) -> find ~time:true ~batch ~fail e
-        | VernacRedirect (_,e) -> find ~time ~batch ~fail e
-        | VernacFail e -> find ~time ~batch ~fail:true e
-        | e -> CAst.make ?loc e, time, batch, fail) v in
-      find ~time:false ~batch:false ~fail:false e in
+    let cl, time, batch, fail =
+      let rec find ~time ~batch ~fail cl = match cl with
+        | ControlTime batch :: cl -> find ~time:true ~batch ~fail cl
+        | ControlRedirect _ :: cl -> find ~time ~batch ~fail cl
+        | ControlFail :: cl -> find ~time ~batch ~fail:true cl
+        | ControlTimeout _ :: cl
```

Comment:
Why are you stripping Timeout? The old code was leaving it these, so that the worker would execute it.

Reply:
Let me see, likely a programming error on my part.

Reply:
The old code is brittle, and the possibility to nest controls as you like makes it very hard to give them a sound meaning. But the new code is just different, and I'm pretty sure I did put in the test suite a `par: Timeout x tac.`

Reply:
Should be fixed now.

Reply:
> I'm pretty sure I did put in the test suite a `par: Timeout x tac.` or `Timeout par: tac` (whatever parses ;-) )

The test-suite passed both ways [dropping the timeout and without], so I dunno but this particular bit didn't have any effect on it.

Reply:
My bad, there are tests with Time, not with Timeout.